include/grpc/impl/codegen/port_platform.h

```diff
@@ -394,6 +394,17 @@
 #endif
 #endif /* GPR_NO_AUTODETECT_PLATFORM */
 
+#if defined(GPR_BACKWARDS_COMPATIBILITY_MODE)
+/*
+ * For backward compatibility mode, reset _FORTIFY_SOURCE to prevent
+ * a library from having non-standard symbols such as __asprintf_chk.
+ * This helps non-glibc systems such as alpine using musl to find symbols.
+ */
+#if defined(_FORTIFY_SOURCE) && _FORTIFY_SOURCE > 0
+#define _FORTIFY_SOURCE 0
```

Comment:
Wouldn't you need to undef it first?

Reply:
Yes, you're right. It should be undefined first.